tests/server/api/test_workers.py

```diff
@@ -465,6 +503,15 @@ async def test_read_work_pools_with_offset(self, client, session):
         result = pydantic.parse_obj_as(List[WorkPool], response.json())
         assert [r.name for r in result] == ["B", "C"]
 
+    async def test_read_work_pool_with_work_pool_that_fails_validation(
+        self,
+        client,
+        invalid_work_pool,
+    ):
+        response = await client.post("/work_pools/filter")
+        assert response.status_code == 200
+        assert len(response.json()) == 4
+
```

Comment:
Would it be good to test that updating `invalid_work_pool` both properly and improperly causes `WorkPoolUpdate` to work as expected?

Reply:
Definitely! But there are some existing base job template validation tests for work pool creation and update:
- [test_create_work_pool_template_validation_missing_variables](https://github.com/PrefectHQ/prefect/blob/main/tests/server/api/test_workers.py#L124)
- [test_create_work_pool_template_validation_missing_nested_variables](https://github.com/PrefectHQ/prefect/blob/main/tests/server/api/test_workers.py#L124)
- [test_update_work_pool_template_validation_missing_variables](https://github.com/PrefectHQ/prefect/blob/main/tests/server/api/test_workers.py#L345)
- [test_update_work_pool_template_validation_missing_nested_variables](https://github.com/PrefectHQ/prefect/blob/main/tests/server/api/test_workers.py#L386)
These tests make sure we're maintaining the existing functionality for those routes.

Reply:
Great! Those seem like they're covering it then